plugins/polio/models/performance_thresholds.py

```diff
@@ -0,0 +1,133 @@
+            return self.none()
+
+        if user.is_authenticated:
+            return self.filter(account=user.iaso_profile.account)
+
+        if app_id is not None:  # leaving the possibility to pass app_id if page needs to be embedded
+            try:
+                project = Project.objects.get_for_user_and_app_id(user, app_id)
+                return self.filter(account=project.account)
+            except Project.DoesNotExist:
+                return self.none()
+
+
+class PerformanceThresholds(SoftDeletableModel):
+    account = models.ForeignKey(Account, on_delete=models.CASCADE, related_name="performance_thresholds", null=False)
+    indicator = models.CharField(max_length=200, blank=False)
+    fail_threshold = models.JSONField(null=False, blank=False)  # a json in the JSON Logic format
+    warning_threshold = models.JSONField(null=False, blank=False)  # a json in the JSON Logic format
+    success_threshold = models.JSONField(null=False, blank=False)  # a json in the JSON Logic format
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    # Managers
+    objects = DefaultSoftDeletableManager.from_queryset(PerformanceThresholdQuerySet)()
+    objects_only_deleted = OnlyDeletedSoftDeletableManager.from_queryset(PerformanceThresholdQuerySet)()
+    objects_include_deleted = IncludeDeletedSoftDeletableManager.from_queryset(PerformanceThresholdQuerySet)()
+
+    class Meta:
+        verbose_name = _("Performance Threshold")
+        verbose_name_plural = _("Performance Thresholds")
+        ordering = ["indicator"]
+        unique_together = ["account", "indicator"]
+
+        indexes = [
+            models.Index(fields=["account"]),
+            models.Index(fields=["indicator"]),
+        ]
+
+    def __str__(self):
+        return f"{self.account.name}-{self.indicator}"
+
+    # make method static for easier testing
+    @staticmethod
+    def is_json_logic_rule(data):
+        if not data:
+            return False
+        keys = data.keys()
+        values = list(data.values())
+        operators = ["==", ">", "<", ">=", "<="]
+        unsupported_keys = [key for key in keys if key not in operators]
+        all_keys_valid = not unsupported_keys
+        correct_array_format = len([value for value in values if (isinstance(value, list) and len(value) == 2)]) == len(
+            keys
+        )
+        correct_reference_format = False
+        correct_argument_format = False
+        if correct_array_format:
+            correct_reference_format = len([value for value in values if (value[0] == {"var": "value"})]) == len(keys)
+
+            correct_argument_format = len(
+                [value for value in values if (value[1] == {"var": "average"} or isinstance(value[1], int))]
+            ) == len(keys)
+
+        if all_keys_valid and correct_array_format and correct_reference_format and correct_argument_format:
+            return True
+
+        return False
```

Comment:
```suggestion
        for key, value in data.items():
            if key not in operators:  # change operators depending on where they're defined
                return False
            if not isinstance(value, list) or len(value) != 2:
                return False
            if value[0] != {"var": "value"}:
                return False
            if value[1] != {"var": "average"} and not isinstance(value[1], int):
                return False
        return True
```

or if you prefer one single `if`:

```suggestion
        for key, value in data.items():
            if (
                    key not in operators
                    or not isinstance(value, list)
                    or len(value) != 2
                    or value[0] != {"var": "value"}
                    or (value[1] != {"var": "average"}
                    and not isinstance(value[1], int))
            ):
                return False
        return True
```

Reply:
I have thought about such a solution, but I wanted to return `False` by default